### ISSUE TITLE
[0.0.38] White screen on `ionic run ios`

### ISSUE THREAD

**Issue by MathieuNls** [driftyco/ionic-app-scripts#288]
#### Short description of the problem:

`ionic run ios` doesn't work anymore on 0.0.38. 

The splash screen appears but is replaced by a white screen.

The problem wasn't present in earlier versions.

#### What behavior are you expecting?

The app to be loaded ;).

**Steps to reproduce:**
1. Install 0.0.38
2. ionic run ios

Here's the produced HTML. All links (i.e. assets/icon/favicon.ico, css, js) are working.

```html
<!DOCTYPE html>
<html lang="en" dir="ltr">
<head>
  <meta charset="UTF-8">
  <title>Ionic App</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
  <meta name="format-detection" content="telephone=no">
  <meta name="msapplication-tap-highlight" content="no">

  <link rel="icon" type="image/x-icon" href="assets/icon/favicon.ico">
  <link rel="manifest" href="assets/manifest.json">
  <meta name="theme-color" content="#4e8ef7">

  <!-- un-comment this code to enable service worker
  <script>
    if ('serviceWorker' in navigator) {
      navigator.serviceWorker.register('service-worker.js')
        .then(() => console.log('service worker installed'))
        .catch(err => console.log('Error', err));
    }
  </script>-->

  <link href="build/main.css" rel="stylesheet">

</head>
<body class="platform-ios platform-cordova platform-webview">

  <!-- Ionic's root component and where the app will load -->
  <ion-app></ion-app>

  <!-- cordova.js required for cordova apps -->
  <script src="cordova.js"></script>

  <!-- The polyfills js is generated during the build process -->
  <script src="build/polyfills.js"></script>

  <!-- The bundle js is generated during the build process -->
  <script src="build/main.js"></script>

</body>
</html>
```

As you can see the ` <ion-app></ion-app>` is empty.

**Which @ionic/app-scripts version are you using?**

0.0.38

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

[stacktraces](http://pastebin.com/uUFPvJaa)

![screen shot 2016-10-31 at 2 51 45 pm](https://cloud.githubusercontent.com/assets/7218861/19867317/d3ca7bb2-9f79-11e6-9526-3e3100beb32f.png)

**Comment by danbucholtz** [comment]
@MathieuNls,

Can you please provide more info? Any error output, etc. We don't have anything to work off of here.

Thanks,
Dan

**Comment by MathieuNls** [comment]
Sorry about that @danbucholtz. 

Here are the js errors;

```js
console-via-logger.js:173Unhandled Promise rejection: (8)
"Failed to load about.html"
"; Zone:"
"<root>"
"; Task:"
"Promise.then"
"; Value:"
"Failed to load about.html"
undefined
(anonymous function)console-via-logger.js:172
opolyfills.js:2:7721
rpolyfills.js:2:8148
ipolyfills.js:2:8165
upolyfills.js:1:7864
(anonymous function)polyfills.js:1:7988
rpolyfills.js:1:6435
console-via-logger.js:173Error: Uncaught (in promise): Failed to load about.html 
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/about.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/dashboard.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/login.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/measure.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/signup.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/time.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/watch.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/tips.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/footer.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/header.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/mobile-error.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/mobile-input.htmlFailed to load resource: The requested URL was not found on this server.
console-via-logger.js:173DEVICE READY FIRED AFTER (2)
705
"ms"
```

it looks like the html files are not copied in www.

**Comment by danbucholtz** [comment]
@MathieuNls,

Can you verify if the files are not copied to `www`?

Does everything work with `ionic serve`?

Can you share the output from `ionic run android/ios` and `ionic serve`?

Thanks,
Dan

**Comment by MathieuNls** [comment]
@danbucholtz,

> Can you verify if the files are not copied to www?

They are not; http://pastebin.com/N7bdrZrm

> Does everything work with ionic serve?

yes

> Can you share the output from ionic run android/ios and ionic serve?

- ionic run ios; http://pastebin.com/uUFPvJaa
- ionic serve; http://pastebin.com/Lpxe8nsY

**Comment by danbucholtz** [comment]
@MathieuNls,

This looks correct to me. Are you sure you don't have an exception in your javascript console.

Thanks,
Dan

**Comment by MathieuNls** [comment]
Yes, 

I have these from my earlier post:

```
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/about.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/dashboard.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/login.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/measure.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/signup.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/time.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/watch.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/tips.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/footer.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/header.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/mobile-error.htmlFailed to load resource: The requested URL was not found on this server.
file:///Users/math/Library/Developer/CoreSimulator/Devices/0AE30687-F2D6-46AA-ACD6-8FC912A54A6B/data/Containers/Bundle/Application/005D3FF1-0255-4E3E-BA59-052A1DDB77CF/Toolwatch.app/www/mobile-input.htmlFailed to load resource: The requested URL was not found on this server.
```

**Comment by danbucholtz** [comment]
What are you doing with those files? Are those templates? Are you using `templateUrl` in your components? Can you share me the typescript code for one of your components with the template url?

Generally, the build process in-lines the template, but perhaps that is not working correctly.

Thanks,
Dan

**Comment by MathieuNls** [comment]
Yes, @danbucholtz, it seems to be related to the in-lines template.

I have compared an old ipa of mine and a new one.

On the old one (month old), main.js for the about page looks like:

```js
AboutPage = __decorate$125([
        Component({ template: /* ion-inline-template */ '\n<header-nav [back]="true"></header-nav>\n\n<ion-content id="about">\n\n<h1> {{ \'about-tw-title\' | translate }} </h1>\n<p> {{ \'about-tw\' | translate }} </p>\n<h1> {{ \'about-why-tw-title\' | translate }} </h1>\n<p> {{ \'about-why-tw\' | translate }} </p>\n<h1> {{ \'about-works-title\' | translate }} </h1>\n<p> {{ \'about-works\' | translate }}  </p>\n\n</ion-content>\n<footer-nav [user]="user"></footer-nav>'
        }), 
        __metadata$19('design:paramtypes', [(typeof (_a = typeof NavParams !== 'undefined' && NavParams) === 'function' && _a) || Object, (typeof (_b = typeof TranslateService !== 'undefined' && TranslateService) === 'function' && _b) || Object])
    ], AboutPage);
    return AboutPage;
    var _a, _b;
}());
```

You can see that the template has been copied from about.html to be inlined.

In the new ipa, generated with 0.0.38 (same behaviour on 0.0.39), the templates are not inlined anymore:

```js 
return t.decorators=[{type:r.G,args:[{templateUrl:"about.html"}]}],
```

and the html files are not copied to www.

You can see the ts component here: https://github.com/Toolwatchapp/tw-mobile/blob/master/src/pages/about/about.ts

**Comment by danbucholtz** [comment]
@MathieuNls,

Okay, cool. Thanks, it looks like you're correct that it is not inlining the template. I will try to figure out why.

Thanks,
Dan

**Comment by MathieuNls** [comment]
Hey @danbucholtz,

If you can give some pointers or a time frame during which you think the regression appeared, I am ready to dig in and propose a fix.

M.

**Comment by danbucholtz** [comment]
@MathieuNls,

I downloaded your repo but I get a ton of build errors referencing a back path and then a `shared` dir. Can you publish your shared dir to `npm` and just link to it normally so I can test and recreate the issue?

I pasted your `about.ts` content into our unit tests to do the inline template and it passed, so I am unclear right now exactly what's going on.

Thanks,
Dan

**Comment by MathieuNls** [comment]
@danbucholtz,

The `shared` dir contains git submodules hosted on Github. 
`git clone --recursive  https://github.com/Toolwatchapp/tw-mobile` will fetch them for you.

M.

**Comment by MathieuNls** [comment]
@danbucholtz ,

I noticed that the issue does not appear with the `--dev --debug` flags:

Here's the `ionic-app-scripts build --dev --debug` version:

```js
var AboutPage = (function () {
    function AboutPage(navParams, translate) {
        this.navParams = navParams;
        this.translate = translate;
        __WEBPACK_IMPORTED_MODULE_2__share_src_app___["a" /* GAService */].screenview("ABOUT");
        this.user = this.navParams.get('user');
    }
    AboutPage = __decorate([
        __webpack_require__.i(__WEBPACK_IMPORTED_MODULE_1__angular_core__["G" /* Component */])({template:/*ion-inline-start:"/home/math/www/twapp/tw-mobile-2rc0/src/pages/about/about.html"*/'\n<header-nav [back]="true"></header-nav>\n\n<ion-content id="about">\n\n<h1> {{ \'about-tw-title\' | translate }} </h1>\n<p> {{ \'about-tw\' | translate }} </p>\n<h1> {{ \'about-why-tw-title\' | translate }} </h1>\n<p> {{ \'about-why-tw\' | translate }} </p>\n<h1> {{ \'about-works-title\' | translate }} </h1>\n<p> {{ \'about-works\' | translate }}  </p>\n\n</ion-content>\n<footer-nav [user]="user"></footer-nav>'/*ion-inline-end:"/home/math/www/twapp/tw-mobile-2rc0/src/pages/about/about.html"*/
        }), 
        __metadata('design:paramtypes', [__WEBPACK_IMPORTED_MODULE_0_ionic_angular__["b" /* NavParams */], __WEBPACK_IMPORTED_MODULE_3_ng2_translate_ng2_translate__["b" /* TranslateService */]])
    ], AboutPage);
    return AboutPage;
}());
```

And here's the `ionic-app-scripts build` version:

```js
    var a = function() {
        function t(t, e) {
            this.navParams = t, this.translate = e, o.a.screenview("ABOUT"), this.user = this.navParams.get("user")
        }
        return t.decorators = [{
            type: r.G,
            args: [{
                templateUrl: "about.html"
            }]
        }], t.ctorParameters = [{
            type: i.b
        }, {
            type: s.b
        }], t
    }
```

**Comment by MathieuNls** [comment]
@danbucholtz ,

I finally figured out what was going on... And turns out I wasted your time. Sorry about that...

Back in beta, the build process wasn't working properly with ionic remote build (`ionic package build ios`) and someone on the forum suggested to copy past the content of `app/main.dev.ts` into `app/main.prod.ts`. It did fix the issue at the time and I was able to use ionic remote build.

When I migrated to rc0 and rc1 I did not recall this and leave `app/main.prod.ts` _patched_. Reverting  `app/main.prod.ts` fixes the template issues and  `ionic package build ios` works too.

Thanks for your help.
